truss/util/path.py

```diff
@@ -137,14 +137,27 @@ def is_ignored(
         bool: True if the path matches any of the ignore patterns (i.e., should be ignored),
             and False otherwise.
     """
-    ignore_spec = pathspec.PathSpec.from_lines(
-        pathspec.patterns.GitWildMatchPattern, patterns
-    )
-
     if base_dir:
         path = path.relative_to(base_dir)
 
-    return ignore_spec.match_file(path)
+    path_str = str(path)
+
+    # Handle directory patterns separately to avoid false matches
+    for pattern in patterns:
+        if pattern.endswith("/"):
```

Comment:
The `patterns` comment above suggests that ones ending in slashes will only match directories, what examples of false matches did you see here? Can we add unit tests to make that more clear?